src/rational/mat_q/cmp.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright © 2023 Sven Moog, Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains implementations for comparison of [`MatQ`].
+
+use super::MatQ;
+use flint_sys::fmpq_mat::fmpq_mat_equal;
+
+impl PartialEq for MatQ {
+    /// Checks if two [`MatQ`] instances are equal. Used by the `==` and `!=` operators.
+    ///
+    /// Parameters:
+    /// - `other`: the other value that is compare against `self`
+    ///
+    /// Returns `true` if the elements are equal, otherwise `false`.
+    ///
+    /// # Example
+    /// ```
+    /// use math::rational::MatQ;
+    /// use std::str::FromStr;
+    ///
+    /// let a1 = MatQ::from_str("[[1/2,2],[3/2,4]]").unwrap();
+    /// let a2 = MatQ::from_str("[[2/4,2],[3/2,4]]").unwrap();
+    /// assert!(a1==a2);
+    ///
+    /// let b = MatQ::from_str("[[1,2],[2,4]]").unwrap();
+    ///
+    /// // These are all equivalent and return false.
+    /// let compared: bool = (a1 == b);
+    /// # assert!(!compared);
+    /// let compared: bool = (&a1 == &b);
+    /// # assert!(!compared);
+    /// let compared: bool = (a1.eq(&b));
+    /// # assert!(!compared);
+    /// let compared: bool = (MatQ::eq(&a1,&b));
+    /// # assert!(!compared);
+    /// ```
+    fn eq(&self, other: &Self) -> bool {
+        unsafe { fmpq_mat_equal(&self.matrix, &other.matrix) != 0 }
+    }
+}
+
+// With the [`Eq`] trait, `a == a` is always true.
+// This is not guaranteed by the [`PartialEq`] trait.
+impl Eq for MatQ {}
+
+/// Test that the [`PartialEq`] trait is correctly implemented.
+#[cfg(test)]
+mod test_partial_eq {
+
+    use super::MatQ;
+    use crate::rational::Q;
+    use std::str::FromStr;
+
+    /// Ensures that different instantiations do not break the equality between matrices
+    #[test]
+    fn equality_between_instantiations() {
+        let a = MatQ::from_str("[[0,1],[0,0]]").unwrap();
```

Comment:
Please add a denominator in at least one of the tests as you instantiate `MatQ` and this is its most important feature.
Especially testworthy is something like 1/2 ==2/4. Does the equality hold in the case of matrices?